pyfstat/snr.py

```diff
@@ -56,21 +56,20 @@ class SignalToNoiseRatio:
 
     def __attrs_post_init__(self):
 
-        self.Tsft = self.detector_states.data[0].deltaT
+        have_noise_weights = self.noise_weights is not None
+        have_sqrtSX = self.assumeSqrtSX is not None
 
-        # FIXME: There has to be a simpler logic for this
-        if self.noise_weights is None:
-            if self.assumeSqrtSX is not None:
-                self.Sinv_Tsft = self.Tsft / self.assumeSqrtSX**2
-            else:
-                raise ValueError(
-                    "Need either `assumeSqrtSX` or `noise_weights` to account for background noise"
-                )
-        elif self.assumeSqrtSX is not None:
+        if have_noise_weights == have_sqrtSX:
             raise ValueError(
-                "Need either `assumeSqrtSX` or `noise_weights` to account for background noise"
+                "Need either `assumeSqrtSX` or `noise_weights` to account for background noise."
             )
 
+        self.Tsft = self.detector_states.data[0].deltaT
+        if not have_noise_weights:
```

Comment:
```suggestion
        if have_sqrtSX:
```

Just to be in principle ready for a non-binary future.

Reply:
I agree. There is a XOR right before this, so `not have_noise_weights` <-> `have_sqrtSX`.